src/btree/bt_compact.c

```diff
@@ -103,6 +103,34 @@ __compact_rewrite_lock(WT_SESSION_IMPL *session, WT_REF *ref, bool *skipp)
 	return (ret);
 }
 
+/*
+ * __compact_progress --
+ *     Output a compact progress message.
+ */
+static void
+__compact_progress(WT_SESSION_IMPL *session)
+{
+	WT_BM *bm;
+	uint64_t cur_time, time_diff;
+
+	bm = S2BT(session)->bm;
+	cur_time = __wt_clock(session);
+
+	time_diff = WT_CLOCKDIFF_SEC(cur_time, session->compact->last);
+	if (time_diff > 1) {
```

Comment:
I'd prefer this structure to be more similar to that used for checkpoint progress messages:
https://github.com/wiredtiger/wiredtiger/blob/develop/src/txn/txn_ckpt.c#L470

Including matching how often messages are generated